dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -888,8 +937,9 @@ boolean handleOutliers(ThroughputDistribution distribution, double throughput) {
                             Integer largestPoolSize = getLargestValidPoolSize(poolSize, forecast, upwardCompareSpan);
                             // only make this check if data is available well beyond compareSpan
                             if ((largestPoolSize - poolSize) > (upwardCompareSpan * compareSpanRatioMultiplier))
-                                if (leanTowardShrinking(poolSize, largestPoolSize, forecast, threadStats.get(largestPoolSize).getMovingAverage()))
-                                    shrinkScore = 0.7;
+                                if (poolSize > hangBufferPoolSize)
```

Comment:
Moved the hangBufferPoolSize check way up to the top of the containing if clause, since it was being checked in all the branches within (oops ... )
            // lean toward shrinking if cpuUtil is high
            if ((shrinkScore < 0.5) && (poolSize > hangBufferPoolSize) && (priorKey > coreThreads)) {

Having the other if statements sequential is clarifying to me, that we do the simple checks first and only call leanTowardShrinking if they pass ... don't really see value in combining them ... is that just a coding style thing? If so, maybe my style is different than yours ... ;-)

Reply:
Code style thing again.  It is the same and doesn't matter that much.